std-bits/table/src/main/java/org/enso/table/data/column/operation/map/numeric/arithmetic/NumericBinaryOpImplementation.java

```diff
@@ -32,325 +45,249 @@ public NumericBinaryOpImplementation(String name) {
-        return switch (storage) {
-          case AbstractLongStorage s -> runLongMap(s, argAsLong, problemAggregator);
-          case BigIntegerStorage s -> runBigIntegerMap(
-              BigIntegerArrayAdapter.fromStorage(s),
-              BigInteger.valueOf(argAsLong),
-              problemAggregator);
-          case BigDecimalStorage s -> runBigDecimalMap(
-              BigDecimalArrayAdapter.fromBigDecimalStorage(s),
-              BigDecimal.valueOf(argAsLong),
-              problemAggregator);
-          case DoubleStorage s -> runDoubleMap(s, (double) argAsLong, problemAggregator);
-          default -> throw new IllegalStateException(
-              "Unsupported storage: " + storage.getClass().getCanonicalName());
-        };
-      } else if (NumericConverter.isCoercibleToDouble(arg)) {
-        double doubleArg = NumericConverter.coerceToDouble(arg);
-        return switch (storage) {
-          case AbstractLongStorage s -> runDoubleMap(
-              DoubleArrayAdapter.fromStorage(s), doubleArg, problemAggregator);
-          case BigIntegerStorage s -> runDoubleMap(
-              DoubleArrayAdapter.fromBigIntegerStorage(s), doubleArg, problemAggregator);
-          case BigDecimalStorage s -> runBigDecimalMap(
-              BigDecimalArrayAdapter.fromBigDecimalStorage(s),
-              BigDecimal.valueOf(doubleArg),
-              problemAggregator);
-          case DoubleStorage s -> runDoubleMap(s, doubleArg, problemAggregator);
-          default -> throw new IllegalStateException(
-              "Unsupported storage: " + storage.getClass().getCanonicalName());
-        };
-      } else if (arg instanceof BigDecimal bd) {
-        return runBigDecimalMap(
-            BigDecimalArrayAdapter.fromAnyStorage(storage), bd, problemAggregator);
-      } else {
-        throw new UnexpectedTypeException("a Number.");
-      }
+      throw new UnexpectedTypeException("a Number.");
     }
   }
 
   @Override
   public Storage<? extends Number> runZip(
       I storage, Storage<?> arg, MapOperationProblemAggregator problemAggregator) {
-    return switch (storage) {
-      case DoubleStorage lhs -> switch (arg) {
-        case BigDecimalStorage rhs -> {
-          BigDecimalArrayAdapter left = BigDecimalArrayAdapter.fromStorage(lhs);
-          BigDecimalArrayAdapter right = BigDecimalArrayAdapter.fromBigDecimalStorage(rhs);
-          yield runBigDecimalZip(left, right, problemAggregator);
-        }
-        default -> runDoubleZip(lhs, DoubleArrayAdapter.fromAnyStorage(arg), problemAggregator);
+    if (storage instanceof ColumnDoubleStorage lhs) {
+      return switch (arg) {
+        case BigDecimalStorage rhs -> runBigDecimalZip(
+            new ColumnStorageFacade<>(lhs, BigDecimal::valueOf), rhs, problemAggregator);
+        case BigIntegerStorage rhs -> runDoubleZip(
+            lhs, new DoubleStorageFacade<>(rhs, BigInteger::doubleValue), problemAggregator);
```

Comment:
We are repeating stuff like 
```
new DoubleStorageFacade<>(rhs, BigInteger::doubleValue)
```
quite a lot across the code.

It's a small piece so not really harmful, but if we ever need to refactor it we'll need to do maaany replacements.

What do you think about having a
```
DoubleStorageFacade.fromBigIntegerStorage(rhs)
```
helper (and analogous ones for BigDecimal and LongStorage)?

I think it's good to have the facade instantiation for each type in a single place. In case we need to tune it (e.g. you decide to switch to a specialization for long->double that avoids boxing) - we'd only have to update a single place versus so many callsites.